application/common/models/User.php

```diff
@@ -641,6 +641,28 @@ public function getNagState()
         return $this->nagState->getState();
     }
 
+    private static function listUsersWithExternalGroupWith($appPrefix): array
```

Comment:
In general, yes: I'm trying to place new functions alphabetically. It's a bit subjective since many of the other functions aren't alphabetical, but it seems to reduce conflicts that would have occurred from multiple people just putting new functions at the end. 🤷‍♂️ 

And regarding the name, my thinking is this: A function named `listSomething()` is getting a list of identifiers of some sort, rather than getting the full objects themselves (vs. `getTheThings()`, for example). I can go with the more specific name (`listUserEmailsWith...`), as that does improve clarity, which is certainly valuable. I'm just trying to balance that against super-long names.

If you have a strong preference, please feel free to say so. For now, I think I'll leave it as-is, though I'm open to changing it.

Reply:
I guess the type system clarifies what is actually returned, so leaving it is fine.